py4j-java/src/main/java/py4j/ClientServer.java

```diff
@@ -119,21 +118,24 @@ public ClientServer(int javaPort, InetAddress javaAddress, int pythonPort, InetA
 	 * @param enableMemoryManagement
 	 */
 	public ClientServer(int javaPort, InetAddress javaAddress, int pythonPort, InetAddress pythonAddress,
-			int connectTimeout, int readTimeout, ServerSocketFactory sSocketFactory, SocketFactory socketFactory,
-			Object entryPoint, boolean autoStartJavaServer, boolean enableMemoryManagement) {
-		this(javaPort, javaAddress, pythonPort, pythonAddress, connectTimeout, readTimeout, sSocketFactory,
-				socketFactory, entryPoint, autoStartJavaServer, enableMemoryManagement, null);
+			int connectTimeout, int readTimeout, int createSocketTimeout, ServerSocketFactory sSocketFactory,
+			SocketFactory socketFactory, Object entryPoint, boolean autoStartJavaServer,
+			boolean enableMemoryManagement) {
+		this(javaPort, javaAddress, pythonPort, pythonAddress, connectTimeout, readTimeout, createSocketTimeout,
+				sSocketFactory, socketFactory, entryPoint, autoStartJavaServer, enableMemoryManagement, null);
```

Comment:
this already takes `connectTimeout`

```suggestion
			int connectTimeout, int readTimeout, ServerSocketFactory sSocketFactory, SocketFactory socketFactory,
			Object entryPoint, boolean autoStartJavaServer, boolean enableMemoryManagement) {
		this(javaPort, javaAddress, pythonPort, pythonAddress, connectTimeout, readTimeout, sSocketFactory,
				socketFactory, entryPoint, autoStartJavaServer, enableMemoryManagement, null);
```